src/shogun/lib/any.h

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2016, Shogun-Toolbox e.V. <shogun-team@shogun-toolbox.org>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  3. Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Written (W) 2016 Sergey Lisitsyn
+ * Written (W) 2016 Sanuj Sharma
+ */
+
+#ifndef _ANY_H_
+#define _ANY_H_
+
+#include <string.h>
+#include <stdexcept>
+#include <typeinfo>
+#include <cxxabi.h>
+
+namespace shogun
+{
+    /** Converts compiler-dependent name of class to
+     * something human readable.
+     * @return human readable name of class
+     */
+    template <typename T>
+    std::string demangledType()
+    {
+        size_t length;
+        int status;
+        char* demangled = abi::__cxa_demangle(typeid(T).name(), nullptr, &length, &status);
+        std::string demangled_string(demangled);
+        free(demangled);
+        return demangled_string;
+    }
+
+    /** @brief An interface for a policy to store a value.
+     * Policy works with a provided memory region and is able
```

Comment:
Can we make this doc a bit better?

What is a "policy" here, what is a "value" here. Define these things.